apis/meta/v1/types.go

```diff
@@ -0,0 +1,37 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+// A reference to an object in the same namespace as the referent.
+type LocalObjectReference struct {
```

Comment:
Is there a reason to use a nested struct here with only one field? Or could we only add a `Name` field into the `SecretKeySelector` struct?

Reply:
Yeah, doesn't seem to be needed in this case, I have removed that struct.